src/services/tracing.ts

```diff
@@ -2,20 +2,96 @@ import {
   SemanticAttributes,
   SemanticResourceAttributes,
 } from "@opentelemetry/semantic-conventions"
+import type * as graphqlTypes from "graphql"
 import { W3CTraceContextPropagator } from "@opentelemetry/core"
 import { NodeTracerProvider } from "@opentelemetry/sdk-trace-node"
-import { getNodeAutoInstrumentations } from "@opentelemetry/auto-instrumentations-node"
+import { HttpInstrumentation } from "@opentelemetry/instrumentation-http"
+import { GraphQLInstrumentation } from "@opentelemetry/instrumentation-graphql"
+import { MongoDBInstrumentation } from "@opentelemetry/instrumentation-mongodb"
+import { IORedisInstrumentation } from "@opentelemetry/instrumentation-ioredis"
+import { GrpcInstrumentation } from "@opentelemetry/instrumentation-grpc"
 import { registerInstrumentations } from "@opentelemetry/instrumentation"
 import { SimpleSpanProcessor } from "@opentelemetry/sdk-trace-base"
 import { JaegerExporter } from "@opentelemetry/exporter-jaeger"
 import { Resource } from "@opentelemetry/resources"
-import { trace, context, propagation, SpanAttributes } from "@opentelemetry/api"
+import {
+  trace,
+  context,
+  propagation,
+  Span,
+  SpanAttributes,
+  SpanStatusCode,
+} from "@opentelemetry/api"
 import { tracingConfig } from "@config/app"
 
 propagation.setGlobalPropagator(new W3CTraceContextPropagator())
 
+const gqlResponseHook = (span: Span, data: graphqlTypes.ExecutionResult) => {
+  if (data.errors && data.errors.length > 0) {
+    span.recordException({
+      name: "graphql.execution.error",
+      message: JSON.stringify(data.errors),
+    })
+    span.setStatus({
+      code: SpanStatusCode.ERROR,
+    })
+    const firstErr = data.errors[0]
```

Comment:
don't quite get why there is a separation on how the first error is being handle and the rest of the array with the loop line 58

Reply:
Its adding the first error twice - once as `graphql.error.type` and once as `graphql.error.0.type`. This is to make the most common case (only 1 error) nicer to search for - but also allows representing the whole array of errors incase there are more.